pkgs/by-name/ne/nekoray/package.nix

```
@@ -0,0 +1,162 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  fetchurl,
+  libsForQt5,
+  cmake,
+  ninja,
+  protobuf,
+  yaml-cpp,
+  zxing-cpp,
+  callPackage,
+  makeDesktopItem,
+  copyDesktopItems,
+
+  v2ray-geoip,
+  v2ray-domain-list-community,
+  sing-geoip,
+}:
+
+let
+  fetchSource =
+    args:
+    fetchFromGitHub (
+      args
+      // {
+        owner = "MatsuriDayo";
+        repo = args.name;
+      }
+    );
+
+  extraSources = {
+    # revs found in https://github.com/MatsuriDayo/nekoray/blob/<version>/libs/get_source_env.sh
+    Xray-core = fetchSource {
+      name = "Xray-core";
+      rev = "01208225ee7e508044cca8eb6776a117bcecd997";
+      hash = "sha256-R66i9MITdE9JlhD4wV0EitKPxyahQqDNpunUxVTmupA=";
+    };
+    sing-box-extra = fetchSource {
+      name = "sing-box-extra";
+      rev = "d31d6da26a51a929349e0d75fd89dccbe20d1268";
+      hash = "sha256-YlzMAff8VOZGyCP7ksjcmoBDHT5llTYwwXIrs+qO5P4=";
+    };
+
+    # revs found in https://github.com/MatsuriDayo/sing-box-extra/blob/<sing-box-extra.rev>/libs/get_source_env.sh
+    sing-box = fetchSource {
+      name = "sing-box";
+      rev = "64f4eed2c667d9ff1e52a84233dee0e2ca32c17e";
+      hash = "sha256-jIg/+fvTn46h6tE6YXtov+ZaBD/ywApTZbzHlT5v4lM=";
+    };
+    sing-quic = fetchSource {
+      name = "sing-quic";
+      rev = "e396733db4de15266f0cfdb43c392aca0759324a";
+      hash = "sha256-un5NtZPRx1QAjwNhXkR9OVGldtfM1jQoNRUzt9oilUE=";
+    };
+    libneko = fetchSource {
+      name = "libneko";
+      rev = "5277a5bfc889ee7a89462695b0e678c1bd4909b1";
+      hash = "sha256-6dlWDzI9ox4PQzEtJNgwA0pXmPC7fGrGId88Zl+1gpw=";
+    };
+  };
+
+  geodata = {
+    "geoip.dat" = "${v2ray-geoip}/share/v2ray/geoip.dat";
+    "geosite.dat" = "${v2ray-domain-list-community}/share/v2ray/geosite.dat";
+    "geoip.db" = "${sing-geoip}/share/sing-box/geoip.db";
+    "geosite.db" = fetchurl {
+      url = "https://github.com/SagerNet/sing-geosite/releases/download/20240331083526/geosite.db";
+      hash = "sha256-LLLdfOsLN3A69aXEXzSTH8Shc+pWtarHT2qy6zifehk=";
+    };
```

Isn't this already packaged?

```suggestion
    "geosite.db" = "${sing-geosite}/share/sing-box/geosite.db";
```

---

Ah, thanks, forgot to check it.